src/OrchardCore/OrchardCore.Lucene.Core/QueryProviders/SimpleQueryStringQueryProvider.cs

```diff
@@ -24,10 +23,12 @@ public Query CreateQuery(ILuceneQueryService builder, LuceneQueryContext context
 
             foreach(var field in fields)
             {
-                if(field.Contains('^', StringComparison.Ordinal) && Single.TryParse(field.Split("^").Last(), out weight))
+                var fieldWeightArray = field.Split("^");
+
+                if(field.Contains('^', StringComparison.Ordinal) && Single.TryParse(fieldWeightArray.Last(), out weight))
```

Comment:
I assume that here I could only do a if fieldWeightArray.Length > 1 too but I feel like what I did is safer.

Reply:
yes, use Length > 1 and `fieldWeightArray[1]`